lib/analysis/modules/ies_enkf.c

```diff
@@ -0,0 +1,1059 @@
+  ies_enkf_config_free( data->config );
+  free( data );
+}
+
+void teccost(matrix_type * W,
+             matrix_type * D,
+             const char * fname,
+             int ens_size,
+             int izone)
+{
+   FILE *fp2;
+   float costJ[100];
+
+   if (izone == 1){
+      fp2 = fopen(fname, "w");
+      fprintf(fp2, "TITLE = \"%s\"\n",fname);
+      fprintf(fp2, "VARIABLES = \"i\" \"Average J\" ");
+      for (int i = 0; i < ens_size; i++)
+         fprintf(fp2, "\"%d\" ",i);
+      fprintf(fp2,"\n");
+   } else {
+      fp2 = fopen(fname, "a");
+   }
+
+   float costf=0.0;
+   for (int i = 0; i < ens_size; i++){
+      costJ[i]=matrix_column_column_dot_product(W,i,W,i)+matrix_column_column_dot_product(D,i,D,i);
+      costf += costJ[i];
+   }
+   costf= costf/ens_size;
+
+   fprintf(fp2,"%2d %10.3f ", izone-1, costf) ;
+   for (int i = 0; i < ens_size; i++)
+      fprintf(fp2,"%10.3f ", costJ[i] ) ;
+   fprintf(fp2,"\n") ;
+   fclose(fp2);
+}
+
+void teclog(matrix_type * W,
+            matrix_type * D,
+            matrix_type * DW,
+            const char * fname,
+            int ens_size,
+            int itnr,
+            double rcond,
+            int nrsing,
+            int nrobs)
+{
+
+   double diff1W = 0.0;
+   double diff2W = 0.0;
+   double diffW;
+   double costfp,costfd,costf;
+   FILE *fp;
+
+   if (itnr == 1){
+      fp = fopen(fname, "w");
+      fprintf(fp, "TITLE = \"%s\"\n",fname);
+      fprintf(fp, "VARIABLES = \"it\" \"Diff1W\" \"Diff2W\" \"J-prior\" \"J-data\" \"J\" \"rcond\" \"Sing\" \"nrobs\"\n");
+   } else {
+      fp = fopen(fname, "a");
+   }
+
+   for (int i = 0; i < ens_size; i++){
+       diffW=matrix_column_column_dot_product(DW,i,DW,i)/ens_size;
+       diff2W+=diffW;
+       if (diffW > diff1W) diff1W=diffW ;
+   }
+   diff2W=diff2W/ens_size;
+
+   costfp=0.0;
+   costfd=0.0;
+   for (int i = 0; i < ens_size; i++){
+       costfp += matrix_column_column_dot_product(W,i,W,i);
+       costfd += matrix_column_column_dot_product(D,i,D,i);
+   }
+   costfp=costfp/ens_size;
+   costfd=costfd/ens_size;
+   costf= costfp + costfd;
+   fprintf(fp," %d %12.5e %12.5e %12.5e %12.5e %12.5e %12.5e %d %d\n", itnr-1, diff1W, diff2W, costfp, costfd, costf, rcond, nrsing , nrobs ) ;
+   fclose(fp);
+}
+
+void tecfld(matrix_type * W,
+           const char * fname,
+           const char * vname,
+           int ii,
+           int jj,
+           int izone)
+{
+   FILE *fp2;
+   int ic;
+
+   if (izone == 1){
+      fp2 = fopen(fname, "w");
+      fprintf(fp2, "TITLE = \"%s\"\n",fname);
+      fprintf(fp2, "VARIABLES = \"i-index\" \"j-index\" \"%s\"\n",vname);
+      fprintf(fp2, "ZONE F=BLOCK, T=\"Iteration %d\", I=%d, J=%d, K=1\n",izone, ii, jj);
+      ic=0;
+      for (int j=0;j<ii;j++){
+         for (int i=0;i<jj;i++){
+            ic=ic+1;
+            fprintf(fp2,"%d ",i);
+            if (ic==30){
+               ic=0;
+               fprintf(fp2,"\n");
+            }
+         }
+      }
+
+      ic=0;
```

Comment:
Here you are reusing the variable `ic` for two different (although similar ...) purposes; such variable reuse is in my opinion bad form. I know matters are divided on this, but personally I like to create lexical scopes for situations like this:

```C
{
    int ic;
    for (int i, ...
         for (int j, ... 
}

{
    int ic;
    for (int j, ...
          for (int j, ...
}
```
no big deal for such a simple example as this function - but I think it is a good principle to avoid variable reuse.